### PR TITLE
fix: fsnotify watcher panics when closing

### PR DESCRIPTION
**Pull request by wlynxg**
fix https://github.com/gogf/gf/issues/3396

### CODE REVIEW

File: os/gres/testdata/data/data.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package data
...
...rKzS2OTV/dS2XaWdbaIO7cE8X24zb7nT1hErdHjuMzvdaAfjVF8aLLhca6hpccU6LefL+aaF8KZ/yFDw0mtoSDYbbiVRv2f6O/rK/5EOTYWObOVsVX3byOnmnzBsZHQLnDQraHho5zRDM1hTls/+Swyc+rl4OowcDxqZ4oHh3Z2HR5bjmoOinLGaHdP/2qm1AlBNXM3J+Qv9wAwabeKDyQmfw6CWuCIHgyaRuGBgUjRggWzUQoWxwAojwnAoVEV5Sj4L0LOTnRZQSjHB1wiaNoKboge5nEqKiRwMGjNigYFlzILB40oLuUEHc2NqhyfLJi3VB9WZS9EL+gDNEMF9OAG5nEo2iRwMGh6C+9A6CwYPIS3dB3l6QJ44ou4DK8wHd3pAJXEEVw8NB8GteAhHoJQ4IgeDBoHgYDwMYLF00dJr84KCwZJEcDnQoA9cTi4cgVKSiBwMGumBg21iBIslhZZe21koGCwTBJcDje3wwuSUwREoZILIsaD5HDgWmgksEvtZqLKVsMqioVjk+R6yfQqSvYGfTW/ngVDK95DjQbM1cDxdZrCE/M7Si8wnw4Plc+CioGkYfpioX/NAKORzyOGg+RU43A0WsHigZqFNiQm2KW1nBZQiMHA10C/+8BPTd/ZyihEYchxo7ASO8wmGQ55xIceBZkk4YThWbIB6ZIUcBpoMgT/kEGEwFGIo5FDQ5AY3DEqTHSwYLFloqZhhS9UPRVrSWsG9kV4JqOc+yGGgYQs4TAwUhjzYMU8NJD8Bh1nJAajnNMhhoEEFuL1BUBgK0YulIw2QIZGnKpZ+T9lwAkqBiaU+9NziBJQCE/A6oPGFVbA6miGXUwhMLKSDBabDmgtQzj6QnV2QaSDc0RswgPnZB3Ik6BSOA4YUKwCozhKX/gDmJAjI53dwAdDR2mqYgERBsOj8biFneWDO8gqBxYZzcGHQuZgITJgNFShqw7l5L46QSZcQ/H6ghkxpOjLv/IIMreCwvhvB0odm5LDQEREcll4YLH3+tfSV6qUMi6a6UtBpEFyhoghY+rSJHBY6+IHD3qEGuxQ/oTMcOCzXJrD0GdJCfq6B+XmDGizZOAguEzqr2QST2bMQHqVxEDk0dC4Dh7beDP5s9LPQRsQG24gqyKAhGxI9w9QfqQAV0MgFwErRqd/+LwAA//+ohVRS+zcAAA=="); err != nil {
```
Comment (gqcn) [inline os/gres/testdata/data/data.go:6]: do not update the testdata of package res.

File: os/gfsnotify/gfsnotify_watcher_loop.go [hunk 2]
```diff
@@ -20,12 +21,12 @@ func (w *Watcher) watchLoop() {
 	go func() {
 		for {
 			select {
-			// Close event.
-			case <-w.closeChan:
```
Comment (gqcn) [inline os/gfsnotify/gfsnotify_watcher_loop.go:24]: do not remove this codes block, it stops loop when watcher closed.

File: os/gfsnotify/gfsnotify_watcher.go [hunk 3]
```diff
@@ -110,7 +110,6 @@ func (w *Watcher) Close() {
 	if err := w.watcher.Close(); err != nil {
 		intlog.Errorf(context.TODO(), `%+v`, err)
 	}
-	close(w.closeChan)
```
Comment (gqcn) [inline os/gfsnotify/gfsnotify_watcher.go:113]: should be kept.

File: os/gfsnotify/gfsnotify.go [hunk 4]
```diff
@@ -87,7 +86,6 @@ func New() (*Watcher, error) {
 		cache:     gcache.New(),
 		events:    gqueue.New(),
 		nameSet:   gset.NewStrSet(true),
-		closeChan: make(chan struct{}),
```
Comment (gqcn) [inline os/gfsnotify/gfsnotify.go:90]: should be kept.

**Review by gqcn — CHANGES_REQUESTED** [review]

File: os/gres/testdata/data/data.go (hunk 1 above)
Comment (wlynxg) [reply os/gres/testdata/data/data.go:6]: sorry, this should be a change caused by me accidentally triggering gofmt

**Pull request closed without merging**